src/features/offline-checks/model/services/MediaLookupService.ts

```diff
@@ -0,0 +1,87 @@
+import { LookupMediaInput } from '@app/abstract/lib';
+import { ActivityDetails, ActivityModel } from '@app/entities/activity';
+import { ActivityResponse, AppletDetailsResponse } from '@app/shared/api';
+import {
+  getActivityDetailsKey,
+  getAppletDetailsKey,
+  getDataFromQuery,
+} from '@app/shared/lib';
+
+const createMediaLookupService = () => {
+  const lookupInMarkdown = (message: string): boolean => {
+    const videoFound = message.includes('<video');
+    const audioFound = message.includes('<audio');
+    const youtubeFound = message.includes('youtu');
+
+    return videoFound || audioFound || youtubeFound;
+  };
+
+  const lookupInActivity = (activity: ActivityDetails): boolean => {
+    for (let item of activity.items) {
+      if (lookupInMarkdown(item.question)) {
+        return true;
+      }
+
+      if (item.inputType === 'AudioPlayer') {
+        return true;
+      }
+    }
+    return false;
+  };
```

Comment:
We can use [Array's .some](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) for more clarity.

```
return activity.items.some(item => item.inputType === 'AudioPlayer' || lookupInMarkdown(item.question))
```